### DISCUSSION TITLE
Is API for "auto-generate release notes" available?

### DISCUSSION THREAD

**Discussion by pllim** [community/community#10038] [Apps, API and Webhooks] [answered]
https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#creating-automatically-generated-release-notes-for-a-new-release

I am thinking that if API to this "auto-generate release notes" button is available, I don't have to wait till GitHub Release time to generate the change log and compile new contributors manually. In fact, maybe I can tie this to Actions to have it autogenerated *before* tagging a new release.

Is this API accessible? Thanks!

**Comment by MylesBorins** [comment, marked as answer]
https://docs.github.com/en/rest/reference/releases#generate-release-notes-content-for-a-release

You can also pass `generate_release_notes: true` when creating a release via the API

https://docs.github.com/en/rest/reference/releases#create-a-release

**Reply by pllim** [reply]
Thanks!

**Reply by wallentx** [reply]
> https://docs.github.com/en/rest/reference/releases#generate-release-notes-content-for-a-release
> 
> You can also pass `generate_release_notes: true` when creating a release via the API
> 
> https://docs.github.com/en/rest/reference/releases#create-a-release

It would be useful to execute this with `"Contents" repository permissions (read)`

**Comment by pllim** [comment]
Does the "tag_name" value have to be a proper tag or can it be an untagged hash? Let's say, I want to preview the release notes before I tag a release?
